Jwt token authorization implementation

Max-Homishchak opened the pull request.

mkhomishchak committed:
Jwt token authorization implementation

DendeberiaOleksandr commented on src/main/java/com/khomishchak/cryproportfolio/configurations/JwtFilterConfig.java:
```diff
@@ -0,0 +1,71 @@
+            HttpServletResponse response,
+            FilterChain filterChain) throws ServletException, IOException {
+
+        String jwtTokenHeader = request.getHeader(AUTHORIZATION_HEADER);
```
FYI: `There is a static constant org.springframework.http.HttpHeaders.AUTHORIZATION`

DendeberiaOleksandr commented on src/main/java/com/khomishchak/cryproportfolio/configurations/JwtFilterConfig.java:
```diff
@@ -0,0 +1,71 @@
+            return;
+        }
+
+        String token = jwtTokenHeader.substring(7);
```
```suggestion
        String token = jwtTokenHeader.substring("Bearer ".length());
```

mkhomishchak committed:
Updated jwt authorization implementation

fixed compare of current and exp dates
added already existing constant
updated filterChain
replaced and renamed filter component

DendeberiaOleksandr commented:
G
